sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeFixedWidthAggregationMap.java

```diff
@@ -225,4 +232,93 @@ public void printPerfMetrics() {
     System.out.println("Total memory consumption (bytes): " + map.getTotalMemoryConsumption());
   }
 
+  /**
+   * Sorts the key, value data in this map in place, and return them as an iterator.
+   *
+   * The only memory that is allocated is the address/prefix array, 16 bytes per record.
+   */
+  public KVIterator<UnsafeRow, UnsafeRow> sortedIterator() {
+    int numElements = map.numElements();
+    final int numKeyFields = groupingKeySchema.size();
+    TaskMemoryManager memoryManager = map.getTaskMemoryManager();
+
+    UnsafeExternalRowSorter.PrefixComputer prefixComp =
+      SortPrefixUtils.createPrefixGenerator(groupingKeySchema);
+    PrefixComparator prefixComparator = SortPrefixUtils.getPrefixComparator(groupingKeySchema);
+
+    final BaseOrdering ordering = GenerateOrdering.create(groupingKeySchema);
+    RecordComparator recordComparator = new RecordComparator() {
+      private final UnsafeRow row1 = new UnsafeRow();
+      private final UnsafeRow row2 = new UnsafeRow();
+
+      @Override
+      public int compare(Object baseObj1, long baseOff1, Object baseObj2, long baseOff2) {
+        row1.pointTo(baseObj1, baseOff1 + 4, numKeyFields, -1);
```

Comment:
The `-1` here is for `sizeInBytes`; if we needed to, I guess we could retrieve the size in bytes since we know where it's stored relative to the row address.